test/externalTests.sh

```diff
@@ -28,88 +28,227 @@
+  find contracts test -name '*.sol' -type f -print0 | xargs -0 sed -i -e 's/pragma solidity [\^0-9\.]*/pragma solidity >=0.0/'
+  echo "Replaced fixed-version pragmas..."
+}
+
+function replace_libsolc_call
+{
+  # Change "compileStandard" to "compile" (needed for pre-5.x Truffle)
+  sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
+}
+
+function find_truffle_config
+{
+  local config_file="truffle.js"
+  local alt_config_file="truffle-config.js"
+
+  if [ ! -f "$config_file" ] && [ ! -f "$alt_config_file" ]; then
+    echo "No matching Truffle config found."
+  fi
+  if [ ! -f "$config_file" ]; then
+    config_file=alt_config_file
+  fi
+  echo "$config_file"
+}
+
+function force_solc_truffle_modules
+{
+  # Replace solc package by v0.5.0
+  for d in node_modules node_modules/truffle/node_modules
+  do
+  (
+    if [ -d "$d" ]
+    then
+      cd $d
+      rm -rf solc
       git clone --depth 1 -b v0.5.0 https://github.com/ethereum/solc-js.git solc
-      SOLCVERSION="UNDEFINED"
-
-      cd solc
-      npm install
-      cp "$SOLJSON" soljson.js
-      SOLCVERSION=$(./solcjs --version)
-      cd ..
-      echo "Using solcjs version $SOLCVERSION"
-
-      if [ -n "$branch" ]
-      then
-        echo "Cloning $branch of $repo..."
-        git clone --depth 1 "$repo" -b "$branch" "$DIR/ext"
-      else
-        echo "Cloning $repo..."
-        git clone --depth 1 "$repo" "$DIR/ext"
-      fi
-      cd ext
-      echo "Current commit hash: `git rev-parse HEAD`"
-      npm ci
-      # Replace solc package by v0.5.0
-      for d in node_modules node_modules/truffle/node_modules
-      do
-      (
-        if [ -d "$d" ]
-        then
-          cd $d
-          rm -rf solc
-          git clone --depth 1 -b v0.5.0 https://github.com/ethereum/solc-js.git solc
-          cp "$SOLJSON" solc/soljson.js
-        fi
-      )
-      done
-      if [ "$name" == "Zeppelin" -o "$name" == "Gnosis" ]; then
-        echo "Replaced fixed-version pragmas..."
-        # Replace fixed-version pragmas in Gnosis (part of Consensys best practice)
-        find contracts test -name '*.sol' -type f -print0 | xargs -0 sed -i -e 's/pragma solidity [\^0-9\.]*/pragma solidity >=0.0/'
-      fi
-      # Change "compileStandard" to "compile" (needed for pre-5.x Truffle)
-      sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
-      # Force usage of correct solidity binary (only works with Truffle 5.x)
-      cat >> truffle*.js <<EOF
-module.exports['compilers'] = {solc: {version: "$DIR/solc"} };
+      cp "$1" solc/soljson.js
+    fi
+  )
+  done
+}
+
+function force_solc
+{
+  local config_file="$1"
+  local dir="$2"
+
+  cat >> "$config_file" <<EOF
+module.exports['compilers'] = {solc: {version: "$dir/solc"} };
 EOF
+}
+
+function force_solc_settings
+{
+  local config_file="$1"
+  local settings="$2"
+  local evmVersion="$3"
+
```

Comment:
Can you add a log statement such that we know what the tests are run against currently?

Reply:
Yes, sure!